crates/sui-graphql-rpc/src/context_data/sui_sdk_data_provider.rs

```diff
@@ -288,21 +273,7 @@ impl DataProvider for SuiClient {
             .map(SerdeBigInt::from);
 
         let pg = self.read_api().get_checkpoints(after, count, false).await?;
-        let system_state = self.governance_api().get_latest_sui_system_state().await?;
-        let protocol_configs = self.fetch_protocol_config(None).await?;
-
-        let data: Result<Vec<_>, _> = pg
-            .data
-            .iter()
-            .map(|c| convert_json_rpc_checkpoint(c, &system_state, &protocol_configs))
-            .collect();
-
-        let checkpoints = data.map_err(|e| {
-            Error::Internal(format!(
-                "Cannot convert the JSON RPC checkpoint into GraphQL checkpoint type: {}",
-                e.message
-            ))
-        })?;
+        let checkpoints: Vec<Checkpoint> = pg.data.iter().map(Checkpoint::from).collect();
```

Comment:
nit (optional): technically the `Checkpoint` in the `Vec` is unnecessary (common to omit to avoid repetition on the line).

```suggestion
        let checkpoints: Vec<_> = pg.data.iter().map(Checkpoint::from).collect();
```